pkg/endpointmanager/manager.go

```diff
@@ -474,10 +476,24 @@ func (mgr *EndpointManager) AddHostEndpoint(ctx context.Context, owner regenerat
 		return err
 	}
 
+	epLabels := labels.Labels{}
+	epLabels.MergeLabels(labels.LabelHost)
+
+	if k8s.IsEnabled() {
+		// Retrieve k8s labels.
+		if k8sNode, err := k8s.GetNode(k8s.Client(), nodeName); err != nil {
```

Comment:
Can be done in a follow-up but it would be better to reuse the information from `retrieveNodeInformation()` to avoid unnecessary apiserver calls.

Reply:
I've added this to my list of follow ups. It looks like a small change but I'll need to test it before sending.